src/main/java/com/powsybl/openloadflow/network/PiModelArray.java

```diff
@@ -146,4 +146,32 @@ public void roundR1ToClosestTap() {
         }
         r1 = Double.NaN;
     }
+
+    @Override
+    public boolean getNewTapPosition(Direction direction) {
```

Comment:
This should be `updateTapPosition`. A getter is not suppose to change internal state.
This is here that the network notification should be fired regading tap position change.
=> so it means that `PiModel` should be part of Network listenable data.

Reply:
Fixed.